src/snowflake/connector/nanoarrow_cpp/ArrowIterator/CArrowTableIterator.cpp

```diff
@@ -15,6 +15,288 @@
+                      "Convert fixed number column to double column, "
+                      "column scale %d, column type id: %d",
+                      scale, columnSchemaView.type);
+        convertScaledFixedNumberColumn_nanoarrow(&columnSchemaView, columnArray,
+                                                 scale);
+      }
+      break;
+    }
+
+    case SnowflakeType::Type::ANY:
+    case SnowflakeType::Type::BINARY:
+    case SnowflakeType::Type::BOOLEAN:
+    case SnowflakeType::Type::CHAR:
+    case SnowflakeType::Type::DATE:
+    case SnowflakeType::Type::REAL:
+    case SnowflakeType::Type::TEXT:
+    case SnowflakeType::Type::VARIANT:
+    case SnowflakeType::Type::VECTOR: {
+      // Do not need to convert
+      break;
+    }
+
+    case SnowflakeType::Type::ARRAY: {
+      switch (columnSchemaView.type) {
+        case NANOARROW_TYPE_STRING: {
+          // No need to convert json encoded array
+          break;
+        }
+        case NANOARROW_TYPE_LIST: {
+          if (columnSchemaView.schema->n_children != 1) {
+            std::string errorInfo = Logger::formatString(
+                "[Snowflake Exception] invalid arrow schema for array items "
+                "expected 1 "
+                "schema child, but got %d",
+                columnSchemaView.schema->n_children);
+            logger->error(__FILE__, __func__, __LINE__, errorInfo.c_str());
+            PyErr_SetString(PyExc_Exception, errorInfo.c_str());
+            break;
+          }
+
+          ArrowSchema* item_schema = columnSchemaView.schema->children[0];
+          ArrowArrayView* item_array = columnArray->children[0];
+          convertIfNeeded(item_schema, item_array);
+          break;
+        }
+        default: {
+          std::string errorInfo = Logger::formatString(
+              "[Snowflake Exception] unknown arrow internal data type(%s) "
+              "for ARRAY data in %s",
+              NANOARROW_TYPE_ENUM_STRING[columnSchemaView.type],
+              columnSchemaView.schema->name);
+          logger->error(__FILE__, __func__, __LINE__, errorInfo.c_str());
+          PyErr_SetString(PyExc_Exception, errorInfo.c_str());
+          break;
+        }
+      }
+      break;
+    }
+    case SnowflakeType::Type::MAP: {
+      switch (columnSchemaView.type) {
+        case NANOARROW_TYPE_STRING: {
+          // No need to convert json encoded data
+          break;
+        }
+        case NANOARROW_TYPE_MAP: {
+          if (columnSchemaView.schema->n_children != 1) {
+            std::string errorInfo = Logger::formatString(
+                "[Snowflake Exception] invalid arrow schema for map entries "
+                "expected 1 "
+                "schema child, but got %d",
+                columnSchemaView.schema->n_children);
+            logger->error(__FILE__, __func__, __LINE__, errorInfo.c_str());
+            PyErr_SetString(PyExc_Exception, errorInfo.c_str());
+            break;
+          }
```

Comment:
I'm comparing the code with CArrowChunkIterator: https://github.com/snowflakedb/snowflake-connector-python/blob/main/src/snowflake/connector/nanoarrow_cpp/ArrowIterator/CArrowChunkIterator.cpp#L440-L444

do we need to add the `case NANOARROW_TYPE_STRING: {` into the CArrowChunkIterator.cpp when handling the `Map` case?

Reply:
Currently the table iterator does no conversion for String columns along with most other primitives:
https://github.com/snowflakedb/snowflake-connector-python/blob/v3.9.0/src/snowflake/connector/nanoarrow_cpp/ArrowIterator/CArrowTableIterator.cpp#L109-L119
This case is basically saying that we trust pyarrow to convert a string correctly by default.